src/python/pants/option/global_options.py

```diff
@@ -400,6 +403,10 @@ def register_bootstrap_options(cls, register):
                   'Format: property=value. Multiple values should be specified as multiple '
                   'occurrences of this flag. Pants itself may add additional platform properties.',
                    type=list, default=[])
+    register('--remote-execution-headers', advanced=True,
+             help='Headers to set on remote execution requests. '
+                  'Format: header=value. Pants itself may add additional headers.',
+             type=list, default=[])
```

Comment:
I know the option gets parsed with the magic `to_map` below but, from a UX perspective, not having this be a `dict` option is a little weird. I would be okay with making this a dict option, and just concatting all the (key, value) pairs together.

Or not, it probably doesn't make a big difference either way.

Reply:
Changed to dict